tests/networkpolicy_test.go

```diff
@@ -1,173 +1,264 @@
-		vmic = createVMICirros(virtClient, tests.NamespaceTestAlternative, map[string]string{})
+			serverVMI = tests.WaitUntilVMIReady(serverVMI, tests.LoggedInCirrosExpecter)
+			clientVMI = tests.WaitUntilVMIReady(clientVMI, tests.LoggedInCirrosExpecter)
+			clientVMIAlternativeNamespace = tests.WaitUntilVMIReady(clientVMIAlternativeNamespace, tests.LoggedInCirrosExpecter)
 
-		vmia = tests.WaitUntilVMIReady(vmia, tests.LoggedInCirrosExpecter)
-		vmib = tests.WaitUntilVMIReady(vmib, tests.LoggedInCirrosExpecter)
-		vmic = tests.WaitUntilVMIReady(vmic, tests.LoggedInCirrosExpecter)
-	})
+			By("Start HTTP server at serverVMI on ports 80 and 81")
+			tests.StartHTTPServerWithLoggedInVMI(serverVMI, 80, false)
+			tests.StartHTTPServerWithLoggedInVMI(serverVMI, 81, false)
 
-	Context("vms limited by Default-deny networkpolicy", func() {
-		var policy *v1network.NetworkPolicy
-
-		BeforeEach(func() {
-			// deny-by-default networkpolicy will deny all the traffice to the vms in the namespace
-			By("Create deny-by-default networkpolicy")
-			networkpolicy := &v1network.NetworkPolicy{
-				ObjectMeta: v13.ObjectMeta{
-					Name: "deny-by-default",
-				},
-				Spec: v1network.NetworkPolicySpec{
-					PodSelector: v13.LabelSelector{},
-					Ingress:     []v1network.NetworkPolicyIngressRule{},
-				},
-			}
-			var err error
-			policy, err = virtClient.NetworkingV1().NetworkPolicies(vmia.Namespace).Create(networkpolicy)
-			Expect(err).ToNot(HaveOccurred())
 		})
 
-		AfterEach(func() {
-			waitForNetworkPolicyDeletion(policy)
-		})
+		Context("and vms limited by Default-deny networkpolicy", func() {
+			var policy *networkv1.NetworkPolicy
 
-		It("[test_id:1511] should be failed to reach vmia from vmib", func() {
-			By("Connect vmia from vmib")
-			assertPingFail(vmib, vmia)
-		})
+			BeforeEach(func() {
+				// deny-by-default networkpolicy will deny all the traffice to the vms in the namespace
+				By("Create deny-by-default networkpolicy")
+				policy = &networkv1.NetworkPolicy{
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: serverVMI.Namespace,
+						Name:      "deny-by-default",
+					},
+					Spec: networkv1.NetworkPolicySpec{
+						PodSelector: metav1.LabelSelector{},
+						Ingress:     []networkv1.NetworkPolicyIngressRule{},
+					},
+				}
+				createNetworkPolicy(policy)
+			})
+
+			AfterEach(func() {
+				waitForNetworkPolicyDeletion(policy)
+			})
+
+			It("[test_id:1511] should fail to reach serverVMI from clientVMI", func() {
+				By("Connect serverVMI from clientVMI")
+				assertPingFail(clientVMI, serverVMI)
+			})
+
+			It("[test_id:1512] should fail to reach clientVMI from serverVMI", func() {
+				By("Connect clientVMI from serverVMI")
+				assertPingFail(serverVMI, clientVMI)
+			})
+			It("[test_id:369] should deny http traffic for ports 80/81 from clientVMI to serverVMI", func() {
+				PingHTTPEventually(clientVMI, serverVMI, 80).ShouldNot(Succeed(), "should deny HTTP traffic from clientVMI go serverVMI with port 80")
```

Comment:
please align PingHTTPEventually with the assertPingFail which have a block of EventuallyWithOffset and after that ConsistentlyWithOffset, in order to check the ping failure was not just a temporal hiccup

Reply:
Add Eventually + Consistently to both.

Reply:
see please
https://github.com/kubevirt/kubevirt/pull/4061#discussion_r488734007